discovery-infra/utils.py

```diff
@@ -368,3 +366,31 @@ def wrapped(*args, **kwargs):
                 raise
         return wrapped
     return decorator
+
+
+@contextmanager
+def file_lock_context(filepath='/tmp/discovery-infra.lock', timeout=300):
+    logging.getLogger('filelock').setLevel(logging.INFO)
+
+    lock = filelock.FileLock(filepath, timeout)
+    try:
+        lock.acquire()
+    except filelock.Timeout:
+        log.info(
+            'Deleting lock file: %s '
+            'since it exceeded timeout of: %d seconds',
+            filepath, timeout
+        )
+        os.unlink(filepath)
+        lock.acquire()
+
+    try:
+        yield
```

Comment:
if action requires more than 30 seconds we will get an exception ?

Reply:
5 minutes not 30 seconds, and not raise an exception but override the lock.
Do you think I should increase the timeout?

Reply:
5 minutes, and not raise exception but override existing lock (see previews answer)